### PR TITLE
Added flagfile to Windows service install

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 63e8d27fccf6464d0631c413866a79fca89017ef (Job results: [301](https://jenkins.osquery.io/job/osqueryPullRequestAudit/301/)) failed the code audit.

File: osquery/main/windows/daemon.cpp [hunk 1]
```diff
@@ -35,6 +37,7 @@ CLI_FLAG(bool,
          false,
          "Uninstall osqueryd.exe from the Windows Service Control Manager");
 
+const std::string kWindowsDefaultFlagsFile = "C:\\ProgramData\\osquery\\osquery.flags";
```
Comment (theopolis) [inline osquery/main/windows/daemon.cpp]: `OSQUERY_HOME "\\osquery.flags"`

File: osquery/main/windows/daemon.cpp [hunk 2]
```diff
@@ -84,14 +87,28 @@ Status installService(const char* const binPath) {
     return Status(1);
   }
 
+  HANDLE flagsFilePtr = nullptr;
+  std::string binPathWithFlagFile = std::string(binPath) + " --flagfile=";
+  std::string flagsFile =
+      FLAGS_flagfile == "" ? kWindowsDefaultFlagsFile : FLAGS_flagfile;
```
Comment (theopolis) [inline osquery/main/windows/daemon.cpp]: `FLAGS_flagfile.empty()` or `FLAGS_flagfile.size()`.

**Comment by facebook-github-bot-5** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2509/files/63e8d27fccf6464d0631c413866a79fca89017ef..55d2d597b9fc6270beccf069ff9edbe61f8a6441)

**Pull request merged**